gdal_subsetter/transform.py

```diff
@@ -149,28 +146,25 @@ def process_item(self, item, source):
                          if 'data' in (v.roles or []))
 
             if self.config.fallback_authn_enabled:
-                input_filename = download(
-                    asset.href,
-                    output_dir,
-                    logger=self.logger,
-                    access_token=None,
-                    cfg=self.config)
-                input_filename = self.rename_file(input_filename, asset.href)
-                                
+                access_token = None
             else:
-                input_filename = download(
-                    asset.href,
-                    output_dir,
-                    logger=self.logger,
-                    access_token=self.message.accessToken,
-                    cfg=self.config)
-                input_filename = self.rename_file(input_filename, asset.href)
+                access_token = self.message.accessToken
+
+            try:
+                temporary_filename = download(asset.href, output_dir,
+                                              logger=self.logger,
+                                              cfg=self.config,
+                                              access_token=access_token)
+            except Exception as exception:
+                raise DownloadError(asset.href, str(exception)) from exception
```

Comment:
This is one of the more important changes - if the download fails this will now use a custom exception (see `exceptions.py`) to ensure all end-user facing messages make it clear that the download failed, even if the error message associated with the failed download is cryptic.